[PT 4] Local storage and settings

KauhareniaNastassia opened the pull request:
Add saving menu items to the local storage if internet connection is not available
Add saving shopping cart to the local storage
Add changing theme mode to settings page
Add changing color theme to settings page
Add links for contacts
Add permissions for links

https://github.com/KauhareniaNastassia/food_delivery_app/assets/104850152/e9ba97d3-7c93-47a5-8bac-7a5cfb62e4f9
https://github.com/KauhareniaNastassia/food_delivery_app/assets/104850152/a5171819-c3ed-4eee-be0b-fed6275ab369

Commit:
[PT-4] Local storage and settings
Add links to contact

Commit:
[PT-4] Local storage and settings
Add local storage for menu list items
Add local storage for shopping cart items
Refactor code

Commit:
[PT-4] Local storage and settings
Add notification for adding cutlery in shopping cart
Add in bloc checking of internet connection
Add notification about internet connection

Commit:
[PT-4] Local storage and settings
Add changing theme mode to settings page
Add changing color theme to settings page
Refactor theme and colors files

Commit:
[PT-4] Local storage and settings
Add logic for changing font size
Add slider for changing font size
Refactor theme and text styles files

Commit:
[PT-4] Local storage and settings
Refactor some problems

Commit:
[PT-4] Local storage and settings
Add refresher
Refactor notifications

Commit:
[PT-4] Local storage and settings
Refactor code

Commit:
[PT-4] Local storage and settings
Add permissions for contact links

SashaPilevich commented on core/lib/core.dart:
```diff
@@ -1,6 +1,12 @@
+export 'package:liquid_pull_to_refresh/liquid_pull_to_refresh.dart';
 export 'config/di/app_di.dart';
 export 'config/di/data_di.dart';
+export 'package:permission_handler/permission_handler.dart';
```
space

SashaPilevich commented on core_ui/lib/src/widgets/app_bar.dart:
```diff
@@ -1,39 +1,19 @@
         return AppBar(
           title: const Text('Food Delivery App'),
           backgroundColor: Theme.of(context).appBarTheme.backgroundColor,
           titleSpacing: Theme.of(context).appBarTheme.titleSpacing,
```
create variable

SashaPilevich commented on core_ui/lib/src/widgets/buttons/primary_button.dart:
```diff
@@ -13,21 +15,30 @@ class PrimaryButton extends StatelessWidget {
-        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 10),
+        padding: const EdgeInsets.symmetric(
+          horizontal: 10,
+          vertical: 10,
```
recomendation: create class for dimensions

SashaPilevich commented on data/lib/providers/settings_local_provider.dart:
```diff
@@ -0,0 +1,45 @@
+
+  Future<double> getFontSizeFromLocal() async {
+    final Box colorScheme = await Hive.openBox('fontSize');
+    final fontSize = colorScheme.get('fontSize').toString();
```
type

SashaPilevich commented on data/lib/repositories/menu_repository_impl/menu_repository_impl.dart:
```diff
@@ -1,24 +1,43 @@
-    return menuItems
-        .map((MenuItemEntity e) => MenuItemMapper.toModel(e))
-        .toList();
+    final ConnectivityResult connectivityResult =
```
create class in core

SashaPilevich commented on features/main_page_view/lib/src/bloc/main_page/bloc.dart:
```diff
@@ -1,37 +1,72 @@
 class MenuBloc extends Bloc<MenuEvent, MenuState> {
   final FetchMenuItemsUseCase _fetchMenuItemsUseCase;
 
   MenuBloc({required FetchMenuItemsUseCase fetchMenuItemsUseCase})
```
,

SashaPilevich commented on features/main_page_view/lib/src/bloc/main_page/state.dart:
```diff
@@ -1,19 +1,30 @@
+    return MenuState(
+      menu: menu ?? this.menu,
+      exception: exception ?? this.exception,
+      isInternetConnectionAvailableState: isInternetConnectionAvailableState ??
```
hasInternetConnection

SashaPilevich commented on features/main_page_view/lib/src/ui/components/main_page_screen.dart:
```diff
@@ -18,42 +19,71 @@ class _MainPageScreenState extends State<MainPageScreen> {
   @override
   Widget build(BuildContext context) {
     final Size size = MediaQuery.of(context).size;
```
final MediaQueryData mediaQueryData = MediaQuery.of(context);

SashaPilevich commented on features/main_page_view/lib/src/ui/components/main_page_screen.dart:
```diff
@@ -18,42 +19,71 @@ class _MainPageScreenState extends State<MainPageScreen> {
+            }
+            if (state.menu.isNotEmpty) {
+              return PageRefresher(
+                onRefresh: () {
```
async

SashaPilevich commented on features/main_page_view/lib/src/ui/components/main_page_screen.dart:
```diff
@@ -18,42 +19,71 @@ class _MainPageScreenState extends State<MainPageScreen> {
+                  return Future<void>(
+                    () => context.read<MenuBloc>().add(
+                          InitEvent(),
+                        ),
```
create variable

SashaPilevich commented on features/main_page_view/lib/src/ui/components/menu_item_details_screen.dart:
```diff
@@ -84,7 +84,7 @@ class _MenuItemDetailsScreenState extends State<MenuItemDetailsScreen> {
                 child: Container(
                   padding: const EdgeInsets.fromLTRB(30, 0, 30, 40),
                   color: Theme.of(context).cardColor,
```
create variable

SashaPilevich commented on features/main_page_view/lib/src/ui/components/widgets/menu_item/item_list_button.dart:
```diff
@@ -19,7 +21,7 @@ class ItemListButtonState extends State<ItemListButton> {
   @override
   Widget build(BuildContext context) {
     final Size size = MediaQuery.of(context).size;
```
same

SashaPilevich commented on features/main_page_view/lib/src/ui/components/widgets/menu_item/menu_item_title.dart:
```diff
@@ -14,6 +16,7 @@ class MenuItemTitle extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final Size size = MediaQuery.sizeOf(context);
```
same

SashaPilevich commented on features/settings/lib/src/bloc/settings_bloc/bloc.dart:
```diff
@@ -0,0 +1,96 @@
+  }
+
+  Future<void> _initAppSettings(
+      InitAppSettings event, Emitter<SettingsState> emit) async {
```
,

SashaPilevich commented on features/settings/lib/src/ui/settings.dart:
```diff
@@ -1,30 +1,52 @@
+                SwitchToTheme(
+                  isLight: state.isLight,
+                  onTap: () {
+                    context.read<SettingsBloc>().add(
```
create bloc variable

SashaPilevich commented on features/shopping_cart/lib/src/bloc/bloc.dart:
```diff
@@ -1,45 +1,78 @@
+        );
+      } else {
+        double totalPrice = 0.0;
+        for (final item in shoppingCartItems) {
```
type

Commit:
[PT-4] Local storage and settings
Refactor code according to review comments

Commit:
[PT-4] Local storage and settings
Refactor code according to review comments

Commit:
[PT-4] Local storage and settings
Refactor code according to review comments

Commit:
[PT-5] Network
Add sign in with firebase auth
Add sign up with firebase auth
Add sign out
Save user info to firebase and local
Create sign in, sign up pages

Commit:
[PT-5] Network
Add sign in via google
Refactor code

Commit:
[PT-5] Network
Change switch between sign in and sign up pages

Commit:
[PT-5] Network
Refactor code

AliaksandrHuzich commented on core/lib/config/di/data_di.dart:
```diff
@@ -52,4 +85,74 @@ class DataDI {
+    );
+  }
+
+  _initSettings() {
```
void

AliaksandrHuzich commented on core/lib/config/internet_connection_check/connection_di.dart:
```diff
@@ -0,0 +1,12 @@
+import 'package:core/core.dart';
```
file naming

AliaksandrHuzich replied:
to data as a provider/service

AliaksandrHuzich commented on core_ui/lib/src/theme/app_theme.dart:
```diff
@@ -2,91 +2,148 @@ import 'package:core_ui/core_ui.dart';
-  );
+  ThemeData get lightThemeData {
+    return ThemeData(
+      scaffoldBackgroundColor: AppColors.backgroundColor,
```
убрать условия сделать 4 темы или тернарным

AliaksandrHuzich commented on core_ui/lib/src/widgets/page_refresher.dart:
```diff
@@ -0,0 +1,30 @@
+
+class PageRefresher extends StatelessWidget {
+  final Widget child;
+  final Future<void> Function() onRefresh;
```
VoidCallback()

AliaksandrHuzich commented on data/lib/entity/shopping_cart_item/shopping_cart_item_entity.dart:
```diff
@@ -0,0 +1,11 @@
+
+class ShoppingCartItemEntity {
+  final MenuItemEntity menuItemEntity;
+  int amount;
```
final

AliaksandrHuzich replied:
for changing - copyWith

AliaksandrHuzich commented on data/lib/providers/local_menu_provider.dart:
```diff
@@ -0,0 +1,24 @@
+import 'package:domain/domain.dart';
+
+class LocalMenuProvider {
+  LocalMenuProvider();
```
const

AliaksandrHuzich commented on data/lib/providers/local_menu_provider.dart:
```diff
@@ -0,0 +1,24 @@
+
+  Future<void> saveMenuItemsToLocal(List<MenuItemModel> menuItems) async {
+    final Box<MenuItemEntity> menuItemsBox = await Hive.openBox('menuItems');
+    if (menuItemsBox.isEmpty) {
```
useless

AliaksandrHuzich commented on data/lib/providers/local_shopping_cart_provider.dart:
```diff
@@ -0,0 +1,62 @@
+import 'package:data/data.dart';
+
+class LocalShoppingCartProvider {
+  LocalShoppingCartProvider();
```
const

AliaksandrHuzich commented on data/lib/providers/local_shopping_cart_provider.dart:
```diff
@@ -0,0 +1,62 @@
+import 'package:data/data.dart';
```
rewrite this file (just a hive provider not a cart one)

AliaksandrHuzich commented on domain/lib/usecases/add_shopping_cart_item_use_case.dart:
```diff
@@ -0,0 +1,13 @@
+  final ShoppingCartRepository _shoppingCartRepository;
+
+  const AddShoppingCartItemUseCase(
+      {required ShoppingCartRepository shoppingCartRepository})
```
,

AliaksandrHuzich commented on features/main_page_view/lib/src/bloc/main_page/bloc.dart:
```diff
@@ -1,37 +1,70 @@
     Emitter<MenuState> emit,
   ) async {
-    emit(MenuLoadingState());
+    add(IsInternetConnectionAvailableEvent());
```
remove

AliaksandrHuzich commented on features/main_page_view/lib/src/bloc/main_page/bloc.dart:
```diff
@@ -1,37 +1,70 @@
   }
+
+  Future<void> _isInternetConnectionAvailable(
+    IsInternetConnectionAvailableEvent event,
```
передай резалт сюда

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/main_page_screen.dart:
```diff
@@ -17,43 +18,70 @@ class _MainPageScreenState extends State<MainPageScreen> {
+        child: BlocConsumer<MenuBloc, MenuState>(
+          listener: (BuildContext context, MenuState state) {
+            if (!state.isInternetConnectionAvailable) {
+              MotionToast.error(
```
to separate widget

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/main_page_screen.dart:
```diff
@@ -17,43 +18,70 @@ class _MainPageScreenState extends State<MainPageScreen> {
+                          const Column(
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            children: <Widget>[
+                              SizedBox(height: 20),
```
todo

AliaksandrHuzich commented on features/settings/lib/src/bloc/open_links_bloc/bloc.dart:
```diff
@@ -0,0 +1,115 @@
+
+class OpenContactLinksBloc
+    extends Bloc<OpenContactLinksEvent, OpenContactLinksState> {
+  OpenContactLinksBloc() : super(OpenContactLinksState()) {
```
formatting

AliaksandrHuzich commented on features/settings/lib/src/bloc/open_links_bloc/bloc.dart:
```diff
@@ -0,0 +1,115 @@
+    }
+  }
+
+  void _openMessengerLink(
```
Future

AliaksandrHuzich commented on features/settings/lib/src/bloc/open_links_bloc/bloc.dart:
```diff
@@ -0,0 +1,115 @@
+    const String telephoneNumber = '+375291234567';
+    final Uri messageUrl = Uri.parse('sms:$telephoneNumber');
+
+    PermissionStatus status = await Permission.sms.request();
```
final

AliaksandrHuzich commented on features/settings/lib/src/ui/components/change_color_scheme.dart:
```diff
@@ -0,0 +1,85 @@
+          style: theme.textTheme.titleLarge,
+        ),
+        GestureDetector(
+          onTap: !isStandardColorScheme ? onTap : () {},
```
развернуть условие

AliaksandrHuzich commented on features/settings/lib/src/ui/components/change_color_scheme.dart:
```diff
@@ -0,0 +1,85 @@
+                      size: size.width * 0.08,
+                      color: Colors.white,
+                    )
+                  : null),
```
,

KauhareniaNastassia committed:
Merge pull request #7 from KauhareniaNastassia/feature/PT-5/network

[PT 5] Network

Commit:
[PT-5] Network
Refactor code according to review

Commit:
Merge branch 'feature/PT-5/network' into feature/PT-4/local-storage_and_settings

Commit:
[PT-5] Network
Refactor code according to review
Add main icon for app
Refactor folder structure
Add total price for item in shopping cart

Commit:
[PT-5] Network
Refactor code
